development/webpack/utils/helpers.ts

```diff
@@ -0,0 +1,243 @@
+  ${solutions.map((solution) => ` •  ${solution}`).join('\n')}
+  ${`\n ${context}`}
+  `;
+
+  throw new Error(message);
+}
+
+/**
+ * It gets minimizers for the webpack build.
+ */
+export function getMinimizers() {
+  const TerserPlugin: typeof TerserPluginType = require('terser-webpack-plugin');
+  return [
+    new TerserPlugin({
+      // use SWC to minify (about 7x faster than Terser)
+      minify: TerserPlugin.swcMinify,
+      // do not minify snow.
+      exclude: /snow\.prod/u,
+    }),
+  ];
+}
+
+/**
+ * Helpers for logging to the console with color.
+ */
+export const { colors, toGreen, toOrange, toPurple } = ((depth, esc) => {
+  if (depth === 1) {
+    const echo = (message: string): string => message;
+    return { colors: false, toGreen: echo, toOrange: echo, toPurple: echo };
+  }
+  // 24: metamask green, 8: close to metamask green, 4: green
+  const green = { 24: '38;2;186;242;74', 8: '38;5;191', 4: '33' }[depth];
+  // 24: metamask orange, 8: close to metamask orange, 4: red :-(
+  const orange = { 24: '38;2;247;85;25', 8: '38;5;208', 4: '31' }[depth];
+  // 24: metamask purple, 8: close to metamask purple, 4: purple
+  const purple = { 24: '38;2;208;117;255', 8: '38;5;177', 4: '35' }[depth];
+  return {
+    colors: { green: `${esc}[1;${green}m`, orange: `${esc}[1;${orange}m` },
+    toGreen: (message: string) => `${esc}[1;${green}m${message}${esc}[0m`,
+    toOrange: (message: string) => `${esc}[1;${orange}m${message}${esc}[0m`,
+    toPurple: (message: string) => `${esc}[1;${purple}m${message}${esc}[0m`,
+  };
+})((process.stderr.getColorDepth?.() as 1 | 4 | 8 | 24) || 1, '\u001b');
+
+/**
+ * Logs a summary of build information to `process.stderr` (webpack logs to
+ * stderr).
+ *
+ * Note: `err` and stats.hasErrors() are different. `err` prevents compilation
+ * from starting, while `stats.hasErrors()` is true if there were errors during
+ * compilation itself.
+ *
+ * @param err - If not `undefined`, logs the error to `process.stderr`.
+ * @param stats - If not `undefined`, logs the stats to `process.stderr`.
+ */
+export function logStats(err?: Error | null, stats?: Stats) {
+  if (err) {
+    console.error(err);
+    return;
+  }
+
+  if (!stats) {
+    // technically this shouldn't happen, but webpack's TypeScript interface
+    // doesn't enforce that `err` and `stats` are mutually exclusive.
+    return;
+  }
+
+  const { options } = stats.compilation;
+  // orange for production builds, purple for development
+  const colorFn = options.mode === 'production' ? toOrange : toPurple;
+  stats.compilation.name = colorFn(`🦊 ${stats.compilation.compiler.name}`);
+  if (options.stats === 'normal') {
+    // log everything (computing stats is slow, so we only do it if asked).
+    console.error(stats.toString({ colors }));
+  } else if (stats.hasErrors() || stats.hasWarnings()) {
+    // always log errors and warnings, if we have them.
+    console.error(stats.toString({ colors, preset: 'errors-warnings' }));
```

Comment:
Note: error and warnings don't omit timing information. I think this is OK because there really should never be errors or warnings.

Note: there are currently warnings because of the way snaps has published some of their packages.